blockdag/dag.go

```diff
@@ -816,20 +816,51 @@ func (dag *BlockDAG) LastFinalityPointHash() *daghash.Hash {
 	return dag.lastFinalityPoint.hash
 }
 
+// isInSelectedParentChain returns whether aNode is in the selected parent chain of bNode.
+func (dag *BlockDAG) isInSelectedParentChain(aNode, bNode *blockNode) (bool, error) {
+	// Because reachabilityInterval.isAncestorOf and isInSelectedParentChain
+	// is exclusive, we need to explicitly check if aNode and bNode
+	// are equal.
+	if aNode == bNode {
+		return false, nil
+	}
+	aTreeNode, err := dag.reachabilityStore.treeNodeByBlockNode(aNode)
+	if err != nil {
+		return false, err
+	}
+
+	bTreeNode, err := dag.reachabilityStore.treeNodeByBlockNode(bNode)
+	if err != nil {
+		return false, err
+	}
+
+	return aTreeNode.interval.isAncestorOf(bTreeNode.interval), nil
+}
+
 // checkFinalityRules checks the new block does not violate the finality rules
-// specifically - the new block selectedParent chain should contain the old finality point
+// specifically - the new block selectedParent chain should contain the old finality point.
 func (dag *BlockDAG) checkFinalityRules(newNode *blockNode) error {
```

Comment:
I'd rename to `checkFinalityPointInSelectedParentChain` to be more explicit regarding what we check (since that's the only finality rule anyway)

Reply:
Or something like `checkFinalityViolation` if you want something shorter, but still don't want to imply there's more then one finality rule.